server/src/main/java/com/defold/extender/builders/CSharpBuilder.java

```diff
@@ -0,0 +1,207 @@
+
+        return f;
+    }
+
+    // https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/?tabs=net8plus%2Cwindows#platformarchitecture-restrictions
+    private static String convertPlatform(String platform) {
+        if (platform.equals("arm64-osx"))       return "osx-arm64";
+        if (platform.equals("x86_64-osx"))      return "osx-x64";
+        if (platform.equals("arm64-android"))   return "android-arm64";
+        if (platform.equals("armv7-android"))   return "android-arm32";
+        if (platform.equals("arm64-win32"))     return "win-arm64";
+        if (platform.equals("x86_64-win32"))    return "win-x64";
+        if (platform.equals("x86-win32"))       return "win-x86";
+        if (platform.equals("x86_64-linux"))    return "linux-x64";
+        if (platform.equals("x86_64-ios"))      return "ios-x64";
+        if (platform.equals("arm64-ios"))       return "ios-arm64";
+        return "unknown_platform";
+    }
+
+    private boolean isWindows(String platform) {
+        if (platform.endsWith("win32"))
+            return true;
+        return false;
+    }
+
+    private File runDotnet(File project, String platform) throws IOException, InterruptedException, ExtenderException {
+
+        String csplatform = convertPlatform(this.platform);
+        String cmd = "dotnet publish --nologo -c Release " + String.format("-r %s ", csplatform);
+        cmd += project.getAbsolutePath();
+
+        List<String> commands = new ArrayList<>();
+        commands.add(cmd);
+        ProcessExecutor.executeCommands(processExecutor, commands); // in parallel
+
+        String libName;
+        if (isWindows(platform))
+            libName = String.format("%s.lib", outputName);
+        else
+            libName = String.format("lib%s.a", outputName);
+
+        File csOutput = new File(this.outputDir, csplatform);
+        File csPublish = new File(csOutput, "publish");
+        File publishLibrary = new File(csPublish, libName);
+
+        FileUtils.moveFile(publishLibrary, outputFile);
+        return outputFile;
+    }
+
+    // Build and return the library file(s)
+    public List<File> build() throws IOException, InterruptedException, ExtenderException {
+
+        File project = writeProject();
+
+        File library = runDotnet(project, this.platform);
+
+        List<File> out = new ArrayList<>();
+        out.add(library);
+        return out;
+    }
+
+    public static List<File> getStaticDependencies(String platform) throws IOException {
+        String csplatform = convertPlatform(platform);
+        String dotnetVersion = readFile(DOTNET_VERSION_FILE).trim();
+
+        Path aotBase = Paths.get(NUGET_PACKAGES, String.format("microsoft.netcore.app.runtime.nativeaot.%s/%s/runtimes/%s/native", csplatform, dotnetVersion, csplatform));
+
+        List<File> out = new ArrayList<>();
+        if (platform.equals("arm64-osx") || platform.equals("x86_64-osx"))
```

Comment:
Haven't had time to verify other platforms yet, as they're been non functional to various degrees.